TASK: 5034 fully completely workspace aware behat tests

mhsdesign opened the pull request:
Resolves: #5034
Replaces: https://github.com/neos/neos-development-collection/pull/5167
Replaces: https://github.com/neos/neos-development-collection/pull/5144

- Remove `CRTestSuiteRuntimeVariables::$currentContentStreamId`
- gets rid of the final usages of `I am in content stream` in the behat tests.
- Additionally our hack `\Neos\ContentRepository\Core\ContentGraphFinder::getByWorkspaceNameAndContentStreamId` is now no longer in use by the tests and can be changed to be not at all exposed.
- Removes deprecated usage `$node->subgraphIdentity->contentStreamId` in `\Neos\ContentRepository\TestSuite\Behavior\...\NodeDiscriminator::fromNode`


**Upgrade instructions**

<!-- 
    Add upgrade instructions for breaking changes. 
    Explain who is affected, what has to be adjusted  
-->

**Review instructions**

<!-- 
    If your change is not explained fully by the first section you can 
    add more details here to help the reviewers understand the change.
    We have to understand what you did, why you did it and how we can 
    verify it works correctly and does no harm.
-->

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

mhsdesign committed:
TASK: Remove deprecated usage `$node->subgraphIdentity->contentStreamId`

mhsdesign committed:
TASK: Fix obsolete declaration of `And I am in content stream`

followup to https://github.com/neos/neos-development-collection/pull/5162

mhsdesign committed:
TASK: Remove dead `I am in content stream`

mhsdesign committed:
TASK: Remove further occurrences of `I am in content stream` where they are obsolete

mhsdesign committed:
TASK: Migrate `ContentStreamForking` tests to not fork explicitly but use workspaces

The implicit forking allows us to use workspaces for assertions on the nodes.

kitsunet commented:
I think I like this direction

mhsdesign commented on Neos.ContentRepository.BehavioralTests/Tests/Behavior/Features/ContentStreamForking/ForkContentStreamWithDisabledNodesWithoutDimensions.feature:
```diff
@@ -56,15 +56,17 @@ Feature: On forking a content stream, hidden nodes should be correctly copied as
 
-    When I am in content stream "user-cs-identifier" and dimension space point {}
+    When I am in workspace "user-test" and dimension space point {}
+    Then I expect the workspace to point to content stream "user-cs-identifier"
     And VisibilityConstraints are set to "withoutRestrictions"
     Then I expect node aggregate identifier "lady-eleonode-rootford" to lead to node user-cs-identifier;lady-eleonode-rootford;{}
```
The `Then I expect the workspace to point to content stream` might be kinda obsolete as the `to lead to` step asserts the content stream id

mhsdesign committed:
TASK: Remove obsolete assertion `Then I expect the workspace to point to content stream "user-cs-identifier"`

This was just a temporal idea

mhsdesign committed:
TASK: Remove `CRTestSuiteRuntimeVariables::$currentContentStreamId` and `I am in content stream`

nezaniel commented:
Imho If the tests for ForkContentStream are removed, the command itself must be removed as well. Otherwise we have untested code paths and I really dislike that.

mhsdesign commented:
We dont remove the tests for `ForkContentStream`, we just test it from a higher perspective. Effectively the test is the same.
But we could surely just inline [the forking logic](https://github.com/neos/neos-development-collection/blob/8d9de7a02669ac0f7531d55e494533556a7b1edb/Neos.ContentRepository.Core/Classes/Feature/ContentStreamCommandHandler.php#L132) in the `CreateWorkspace` command i guess as we wont be needing an api way to just fork a content stream. However if we decide to go down that road i would like to keep that change separate as it will likely also span the `CreateContentStream` and further commands that are now obsolete standalone.

bwaidelich reviewed (approved):
Makes sense to me and I like what you [wrote](https://github.com/neos/neos-development-collection/pull/5169#issuecomment-2211154182).
I agree, that we should mark the low-level commands `@internal` or even remove them to make forking a purely internal concept

mhsdesign committed:
TASK: Declare lower level content stream commands as internal

With the content graph / subgraph operating on workspaces, there is no API way of fetching from an arbitrary content stream by id.
And by not being able to access it we declare creating content streams or doing any low level work with it as internal.

mhsdesign commented:
Okay i did declare the content stream commands as internal:

```
@internal implementation detail, please use the higher level workspace commands instead.
```

do you agree as well on going down this road now @nezaniel?

edit:

changed to

```
 * @internal implementation detail. You must not use this command directly.
 * Direct use may lead to hard to revert senseless state in your content repository.
 * Please use the higher level workspace commands instead.
```
